ethsigner/commandline/src/main/java/tech/pegasys/ethsigner/CommandlineParser.java

```diff
@@ -54,67 +57,56 @@ public boolean parseCommandLine(final String... args) {
     final CommandLine commandLine = new CommandLine(baseCommand);
     commandLine.setCaseInsensitiveEnumValuesAllowed(true);
     commandLine.registerConverter(Level.class, Level::valueOf);
+    commandLine.setOut(outputWriter);
+    commandLine.setErr(errorWriter);
+    commandLine.setExecutionExceptionHandler(this::handleExecutionException);
+    commandLine.setParameterExceptionHandler(this::handleParseException);
 
     for (final SignerSubCommand subcommand : signers) {
       commandLine.addSubcommand(subcommand.getCommandName(), subcommand);
     }
 
-    try {
-      commandLine.parseWithHandlers(new RunLast().useOut(output), new ExceptionHandler<>(), args);
-      return true;
-    } catch (final ParameterException ex) {
-      handleParameterException(ex);
-    } catch (final ExecutionException ex) {
-      commandLine.usage(output);
-    } catch (final TransactionSignerInitializationException | InitializationException ex) {
-      // perform no-op (user output already supplied in ExceptionHandler)
-    } catch (final Exception ex) {
-      LOG.error("EthSigner has suffered an unrecoverable failure", ex);
-      output.println("EthSigner has suffered an unrecoverable failure " + ex.toString());
-    }
-    return false;
+    final int resultCode = commandLine.execute(args);
+    return resultCode == CommandLine.ExitCode.OK;
   }
 
-  private void handleParameterException(final ParameterException ex) {
+  private int handleParseException(ParameterException ex, String[] args) {
```

Comment:
final params?

Reply:
interesting ... this should have been caught by errorprone